benchmark/napi/function_args/napi_binding.c

```diff
@@ -0,0 +1,234 @@
+#include <node_api.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+static napi_value CallWithString(napi_env env, napi_callback_info info) {
+  napi_status status;
+
+  size_t argc = 1;
+  napi_value args[1];
+  status = napi_get_cb_info(env, info, &argc, args, NULL, NULL);
+  assert(status == napi_ok);
+
+  napi_valuetype types[1];
+  status = napi_typeof(env, args[0], types);
+  assert(status == napi_ok);
+
+  assert(types[0] == napi_string);
+  if (types[0] == napi_string) {
+    size_t len = 0;
+    // Get the length
+    status = napi_get_value_string_utf8(env, args[0], NULL, 0, &len);
+    assert(status == napi_ok);
+    char* buf = (char*)malloc(len + 1);
+    status = napi_get_value_string_utf8(env, args[0], buf, len + 1, &len);
+    assert(status == napi_ok);
+    free(buf);
+  }
+
+  return NULL;
+}
+
+static napi_value CallWithArray(napi_env env, napi_callback_info info) {
+  napi_status status;
+
+  size_t argc = 1;
+  napi_value args[1];
+  status = napi_get_cb_info(env, info, &argc, args, NULL, NULL);
+  assert(status == napi_ok);
+
+  napi_value array = args[0];
+  bool is_array = false;
+  status = napi_is_array(env, array, &is_array);
+  assert(status == napi_ok);
+
+  assert(is_array);
+  if (is_array) {
+    uint32_t length;
+    status = napi_get_array_length(env, array, &length);
+    assert(status == napi_ok);
+
+    for (uint32_t i = 0; i < length; ++i) {
+      napi_value v;
+      status = napi_get_element(env, array, i, &v);
+      assert(status == napi_ok);
+    }
+  }
+
+  return NULL;
+}
+
+static napi_value CallWithNumber(napi_env env, napi_callback_info info) {
+  napi_status status;
+
+  size_t argc = 1;
+  napi_value args[1];
+  status = napi_get_cb_info(env, info, &argc, args, NULL, NULL);
+  assert(status == napi_ok);
+
+  napi_valuetype types[1];
+  status = napi_typeof(env, args[0], types);
+  assert(status == napi_ok);
+
+  assert(types[0] == napi_number);
+  if (types[0] == napi_number) {
+    double value = 0.0;
+    status = napi_get_value_double(env, args[0], &value);
+    assert(status == napi_ok);
+  }
+
+  return NULL;
+}
+
+static napi_value CallWithObject(napi_env env, napi_callback_info info) {
+  napi_status status;
+
+  size_t argc = 1;
+  napi_value args[1];
+  status = napi_get_cb_info(env, info, &argc, args, NULL, NULL);
+  assert(status == napi_ok);
+
+  napi_valuetype types[1];
+  status = napi_typeof(env, args[0], types);
+  assert(status == napi_ok);
+
+  assert(types[0] == napi_object);
```

Comment:
You should add `&& argc == 1` into the assertion, to be on par with the V8 version.